### ISSUE TITLE
Unknown000 has been solved!

### ISSUE THREAD

**Issue by Heroesflorian** [Sh0wdown/UnofficialCrusaderPatch#616]
The value of Unknown000 determines the types of flags the AI will place on the Flg 1-10 "troop" spots of an AIV (can be set in the AIV Editor). 

Here's a mapping of values for Unknown000 to types of flags the AI uses in their AIV flag spots: 
10 = flag D
11 = flag C
12 = flag A
13 = flag B

![grafik](https://user-images.githubusercontent.com/12549394/82759406-886ee380-9ded-11ea-823b-266e9d05746f.png)

In the vanilla AIC, Arabian AI characters all have 10 (flag D), some European characters use 12 (flag A), some use 13 (flag B); a value of 11 (flag C) is not used for any AI. 
(Independent of that, at the start of a skirmish game, every player spawns with one flag C attached to their keep.) 

![grafik](https://user-images.githubusercontent.com/12549394/82759612-d0dad100-9dee-11ea-90b9-5248e0de1fbc.png)

I successfully tested changing the Unknown000 value to 11 for some AI, causing it to use type C flags as expected: 
![grafik](https://user-images.githubusercontent.com/12549394/82759551-7ccfec80-9dee-11ea-98c5-9f7442724234.png)

Thus, Unknown000 should be renamed to something like "FlagType" or possibly "BannerType", and a new Enum FlagType with the corresponding values should be created.

**Comment by Lolasik011** [comment]
@Heroesflorian Good News !

**Comment by Heroesflorian** [comment]
Side note: Some other values could be used as well instead of the regular flags... a few ones I found out so far include: 

- 26 (fire explosions)
- 9 (burning flames)
- 15 (decorative skulls)

**Comment by Krarilotus** [comment]
nice :D

So this is good news, just a question at the guys with the assembly knowledge @patel-nikhil for example... could you ensure that there is no other connection to this value?
But yeah, this is great!

**Comment by ilosevsrat** [comment]
THANK YOU for HARD TESTING WORK!!

**Comment by Heroesflorian** [comment]
> could you ensure that there is no other connection to this value?

@Krarilotus what exactly do you mean with that?

**Comment by Panbutt** [comment]
> Side note: Some other values could be used as well instead of the regular flags... a few ones I found out so far include:
> 
> * 8 (oil pouring out with fire)
> * 9 (burning flames)
> * 14 (braziers)
> * 15 (decorative skulls)
> * 21 (steam clouds)
> * 22 (disease clouds)
> * 26 (fire explosions)
> * 28 (ravens flying down)
> * 29 (medium rocks landing in water)
> * 30 (small rocks hitting structures)
> * 31 (small rocks landing in water)
> * 40 (zero gold rising note)
> * 41 (zero popularity rising note)
> * 42 (blank rising note)
> * 43 (blank rising note)
> 
> So far I tested the value range from 0-99 (inclusive).

Do these remain permanently? Like the disease clouds and the burning fire. Asking because it would be cool to have burning fire piles scattered around Caliph's castles to accentuate his obsession with fire.

**Comment by Heroesflorian** [comment]
Yes they do. Can only place them in valid locations for placing flags, though.

**Comment by Xebenkeck** [comment]
Awesome, It's good to finally figure out what this one does.

I would have never guessed flags was what changed with this one.

**Comment by patel-nikhil** [comment]
@Heroesflorian just want to confirm that since this is in AIC, this only applies for AI, correct? And so changing it in AIC will not impact the sprite for human players

**Comment by Heroesflorian** [comment]
@patel-nikhil yes, this is as any other aic variable a value that exists per ai character, and already in vanilla different characters use different flag types for the aiv editor Flg spots. 
Changing the variable for some AI does not influence the player's ability for placing or not placing decoration objects  nor his buildings menu UI for those objects.
